### PR TITLE
ic3* code -- minor changes

### PR DESCRIPTION
**Pull request by ahmed-irfan**

### CODE REVIEW

File: engines/ic3.cpp
Comment (makaimann) [inline engines/ic3.cpp:54]: I have a slight preference for `const auto &`. They're semantically the same but since elsewhere we write `const <type-specifier> &`, I think it's more consistent.

File: engines/ic3.cpp
Comment (makaimann) [inline engines/ic3.cpp:72]: ditto

**Review by makaimann — APPROVED** [review]
This looks great! Thanks for making these improvements -- much cleaner / more efficient. I left a minor comment with a personal preference. I didn't comment on all the relevant spots because there are a lot. Feel free to make the change or not, up to you. Otherwise, I think it's good to go except for a merge conflict after the latest PR got merged.

**Comment by ahmed-irfan** [comment]
Please also take a look at the changes in the propagate method

**Comment by makaimann** [comment]
Yeah I saw that, looks good to me. It should be semantically equivalent, right? You're just editing the vector in place by overriding the first `k` elements and then resizing -- very nice. As long as `constrain_frame` is after popping the solver context (which it is) I think it's all good.

After merging https://github.com/upscale-project/pono/pull/146 then `constrain_frame` in this `propagate` should be called with `false`, right?

File: engines/ic3.cpp
Comment (ahmed-irfan) [reply engines/ic3.cpp]: done in 4a22594

File: engines/ic3.cpp
Comment (ahmed-irfan) [reply engines/ic3.cpp]: done in 4a22594

**Comment by ahmed-irfan** [comment]
@makaimann can you please check the last two commits and merge if everything looks fine.

**Pull request merged**
